tests/standalone_tests/conftest.py

```diff
@@ -787,14 +795,31 @@ def localnodetmp_node_service(rabbitmq_localnodetmp, monetdb_localnodetmp):
     The rabbitmq and monetdb containers have also 'function' scope so this is VERY slow.
     This should be used only when the service should be killed e.g. for testing.
     """
-    node_config_file = LOCALNODETMP_CONFIG_FILE
-    algo_folders_env_variable_val = ALGORITHM_FOLDERS_ENV_VARIABLE_VALUE
-    node_config_filepath = path.join(TEST_ENV_CONFIG_FOLDER, node_config_file)
-    proc = _create_node_service(algo_folders_env_variable_val, node_config_filepath)
+    proc = create_localnodetmp_node_service()
     yield proc
     kill_service(proc)
 
 
+def ensure_localnodetmp_node_service_is_running(node_process):
```

Comment:
Nice idea about the methods, changed it.

I am not sure what you mean though about killing the malfunctioning node service process inside the function.

Reply:
what I meant about killing the process inside the function was that since you check if the process is "zombie" (or "sleepping"), if the process is malfunctioning you could kill it just before return the new process

Reply:
I am not sure why it is like that but "killing" a process actually turns it into a zombie...

I didn't have time to look much further into it but from what I saw when I looked into it, the processes are completely removed when the parent process (pytest) finishes.